File: VB/WindowsApplication73/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid.ViewInfo
Imports DevExpress.XtraPivotGrid
Imports DevExpress.XtraPivotGrid.Data

Namespace WindowsApplication73

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            pivotGridControl1.DataSource = CreateTable(500)
            pivotGridControl1.RetrieveFields()
            pivotGridControl1.Fields("Order").Area = PivotArea.RowArea
            pivotGridControl1.Fields("Type").Area = PivotArea.ColumnArea
            pivotGridControl1.Fields("Number").Area = PivotArea.DataArea
        End Sub

        Private Function CreateTable(ByVal RowCount As Integer) As DataTable
            Dim tbl As DataTable = New DataTable()
            tbl.Columns.Add("Order", GetType(String))
            tbl.Columns.Add("Type", GetType(String))
            tbl.Columns.Add("Number", GetType(Integer))
            tbl.Columns.Add("Date", GetType(Date))
            For i As Integer = 0 To RowCount - 1
                tbl.Rows.Add(New Object() {"Order " & i Mod 7, "Type " & i Mod 4, i * 24 Mod 87, Date.Now.AddDays(i)})
            Next

            Return tbl
        End Function

        Private mousePos As Point = New Point()

        Private viewInfo As BaseViewInfo

        Private offset As Integer = 5

        Private Sub pivotGridControl1_CustomDrawFieldValue(ByVal sender As Object, ByVal e As PivotCustomDrawFieldValueEventArgs)
            If mousePos <> Point.Empty AndAlso e.Bounds.Contains(mousePos) Then
                e.Painter.DrawObject(e.Info)
                Dim rectSize As Size = New Size(12, 12)
                Dim rectPos As Point = New Point(e.Bounds.Right - rectSize.Width - offset, e.Bounds.Top + offset)
                Dim rect As Rectangle = New Rectangle(rectPos, rectSize)
                e.Graphics.FillRectangle(Brushes.Chartreuse, rect)
                e.Handled = True
            End If
        End Sub

        Private Sub pivotGridControl1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs)
            mousePos = Point.Empty
            Dim info As PivotGridHitInfo = pivotGridControl1.CalcHitInfo(e.Location)
            If info.HitTest = PivotGridHitTest.Value Then
                mousePos = e.Location
                Dim newVI As BaseViewInfo = CType(info.ValueInfo.Data, PivotGridViewInfoData).ViewInfo.GetViewInfoAtPoint(e.Location)
                If Not ReferenceEquals(newVI, viewInfo) Then
                    pivotGridControl1.Invalidate(newVI.PaintBounds)
                    If viewInfo IsNot Nothing Then pivotGridControl1.Invalidate(viewInfo.PaintBounds)
                    viewInfo = newVI
                End If
            Else
                If viewInfo IsNot Nothing Then pivotGridControl1.Invalidate(viewInfo.PaintBounds)
                viewInfo = Nothing
            End If
        End Sub

        Private Sub pivotGridControl1_MouseLeave(ByVal sender As Object, ByVal e As EventArgs)
            viewInfo = Nothing
            mousePos = Point.Empty
        End Sub
    End Class
End Namespace

File: CS/WindowsApplication73/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid.ViewInfo;
using DevExpress.XtraPivotGrid;
using DevExpress.XtraPivotGrid.Data;

namespace WindowsApplication73
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            pivotGridControl1.DataSource = CreateTable(500);
            pivotGridControl1.RetrieveFields();
            pivotGridControl1.Fields["Order"].Area = PivotArea.RowArea;
            pivotGridControl1.Fields["Type"].Area = PivotArea.ColumnArea;
            pivotGridControl1.Fields["Number"].Area = PivotArea.DataArea;
        }
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Order", typeof(string));
            tbl.Columns.Add("Type", typeof(string));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { "Order "+ i % 7 , "Type " + i % 4, (i * 24) % 87 , DateTime.Now.AddDays(i) });
            return tbl;
        }
        Point mousePos = new Point();
        BaseViewInfo viewInfo;
        int offset = 5;

        private void pivotGridControl1_CustomDrawFieldValue(object sender, DevExpress.XtraPivotGrid.PivotCustomDrawFieldValueEventArgs e)
        {
            if (mousePos != Point.Empty && e.Bounds.Contains(mousePos))
            {
                e.Painter.DrawObject(e.Info);
                Size rectSize = new Size(12, 12);
                Point rectPos = new Point(e.Bounds.Right - rectSize.Width  - offset, e.Bounds.Top + offset);
                Rectangle rect = new Rectangle(rectPos, rectSize);

                e.Graphics.FillRectangle(Brushes.Chartreuse, rect);
                e.Handled = true;
            }
        }

        private void pivotGridControl1_MouseMove(object sender, MouseEventArgs e)
        {
            mousePos = Point.Empty;
            PivotGridHitInfo info = pivotGridControl1.CalcHitInfo(e.Location);
            if (info.HitTest == PivotGridHitTest.Value)
            {
                mousePos = e.Location;
                BaseViewInfo newVI = ((PivotGridViewInfoData)info.ValueInfo.Data).ViewInfo.GetViewInfoAtPoint(e.Location);
                if (!object.ReferenceEquals(newVI, viewInfo))
                {
                    pivotGridControl1.Invalidate(newVI.PaintBounds);
                    if (viewInfo != null)
                        pivotGridControl1.Invalidate(viewInfo.PaintBounds);
                    viewInfo = newVI;
                }
            }
            else
            {
                if (viewInfo != null)
                    pivotGridControl1.Invalidate(viewInfo.PaintBounds);
                viewInfo = null;
            }
        }

        private void pivotGridControl1_MouseLeave(object sender, EventArgs e)
        {
            viewInfo = null;
            mousePos = Point.Empty;
        }

    }
}
